include/CppInterOp/CppInterOp.h

```diff
@@ -877,13 +877,14 @@ CPPINTEROP_API void Destruct(TCppObject_t This, TCppScope_t type,
 enum CaptureStreamKind : char {
   kStdOut = 1, ///< stdout
   kStdErr,     ///< stderr
+  kCustomPipe
   // kStdBoth,    ///< stdout and stderr
   // kSTDSTRM  // "&1" or "&2" is not a filename
 };
 
 /// Begins recording the given standard stream.
 ///\param[fd_kind] - The stream to be captured
-CPPINTEROP_API void BeginStdStreamCapture(CaptureStreamKind fd_kind);
+CPPINTEROP_API void BeginStdStreamCapture(CaptureStreamKind fd_kind, int custom_fd = -1, FILE* external_temp_file = nullptr);
```

Comment:
I would prefer this

```suggestion
/// Begins recording the given standard stream.
///\param[fd] - The stream to be captured
CPPINTEROP_API void BeginStdStreamCapture(int fd);
```

Also, why use `external_temp_file`, when you are going to return a string.

Now, I am not sure about some things myself.
Let's wait till tomorrow's meeting to continue work here.